apps/test/unit/storage/dataBrowser/dataVisualizer/VisualizerModalTest.js

```diff
@@ -225,7 +248,36 @@ describe('VisualizerModal', () => {
         wrapper.instance().filterRecords(records, 'filterCol', '0')
       ).to.deep.equal([records[4]]);
     });
+    it('operators work with numbers', () => {
+      let records = [
+        {id: 1, filterCol: 123, chartCol: 2},
+        {id: 2, filterCol: 456, chartCol: 3},
+        {id: 3, filterCol: 123, chartCol: 5},
+        {id: 4, filterCol: '456', chartCol: 7},
+        {id: 5, filterCol: 0, chartCol: 5}
+      ];
+      expect(
+        wrapper.instance().filterRecords(records, 'filterCol', '123', '>')
+      ).to.deep.equal([records[1], records[3]]);
 
+      expect(
+        wrapper.instance().filterRecords(records, 'filterCol', '456', '<')
+      ).to.deep.equal([records[0], records[2], records[4]]);
+
+      expect(
+        wrapper.instance().filterRecords(records, 'filterCol', '456', '<=')
+      ).to.deep.equal([
+        records[0],
+        records[1],
+        records[2],
+        records[3],
```

Comment:
It looks like records[3] is a string, so I would expect that to not show up here. Does our system automatically cast strings to numbers?

Reply:
Ah, interesting! I thought it was intentional because we coerce the type of the _filter_ value [here](https://github.com/code-dot-org/code-dot-org/blob/880be3183e7ef622eba5ffd79b75c22e0c1e28cd/apps/src/storage/dataBrowser/dataVisualizer/VisualizerModal.jsx#L122). However, I see that the ["works with numbers" test](https://github.com/code-dot-org/code-dot-org/blob/880be3183e7ef622eba5ffd79b75c22e0c1e28cd/apps/test/unit/storage/dataBrowser/dataVisualizer/VisualizerModalTest.js#L220) expects that we don't treat strings and values equally, and we are using `===` for the test. So I believe we do _not_ want to cast strings to numbers for this. It gets a little funky in the code because the comparison operators like `>=` automatically coerce strings to numbers. So I'll update this test and will need to add type checks to the filtering logic!

Reply:
Oh, I see - I remember there being discussions about how these types are determined. @ajpal - I believe you have more context knowledge here - can you chime in on whether we do expect the cast-ing from strings to numbers as seen here?

Reply:
hmm, I think the behavior as implemented is correct, but I'm not sure I see the value of this test case. Since it has a string in it, `filterCol` is not a numeric column, and so can only be filtered using exact match still. So filtering the records with `<=` I think should be an unreachable state?

Reply:
@ajpal @epeach  Ah that makes sense. You're right that the filter options should never be available for that column.

Given that, I think the best thing to do is remove the type-checking behavior in the latest commit and omit the string value from the test since this path should be unreachable for it. Does that make sense?